spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/repository/query/DatastoreQueryLookupStrategy.java

```diff
@@ -39,17 +39,17 @@ public class DatastoreQueryLookupStrategy implements QueryLookupStrategy {
 
   private final DatastoreMappingContext datastoreMappingContext;
 
-  private QueryMethodEvaluationContextProvider evaluationContextProvider;
+  private final ValueExpressionDelegate valueExpressionDelegate;
```

Comment:
> spring-cloud-gcp-data-datastore: Upgrade [deprecated](https://github.com/spring-projects/spring-data-commons/commit/096836ee808d2cf82caf2efd48cb56271e6ee7c8) QueryMethodEvaluationContextProvider to ValueExpressionDelegate

It looks like we're escalating a deprecation by Spring Data into a breaking change in Spring Cloud GCP. Do we have to drop support for QueryMethodEvaluationContextProvider, or can we support both and just "pass on" the deprecation warning to the user?

Same question applies to `spring-cloud-gcp-data-spanner`.

Reply:
I though this new ValueExpressionDelegate was something internally handled via bean injection. I'll add support for both.